server/src/main/java/org/apache/iotdb/db/metadata/idtable/deviceID/StandAloneAutoIncDeviceID.java

```diff
@@ -0,0 +1,310 @@
+    synchronized (deviceIDs) {
+      if (id.autoIncrementID < deviceIDs.size() && deviceIDs.get(id.autoIncrementID) != null) {
+        return (StandAloneAutoIncDeviceID) deviceIDs.get(id.autoIncrementID);
+      } else {
+        for (int i = deviceIDs.size(); i < id.autoIncrementID; i++) {
+          deviceIDs.add(i, null);
+        }
+        deviceIDs.add(id.autoIncrementID, id);
+        return id;
+      }
+    }
+  }
+
+  /**
+   * get device id from a standAloneAutoIncDeviceID
+   *
+   * @param deviceID StandAloneAutoIncDeviceID deviceID, like: "`1`"
+   * @return a standAloneAutoIncDeviceID instance
+   */
+  private static StandAloneAutoIncDeviceID fromAutoIncDeviceID(String deviceID) {
+    long id = parseFromDeviceID(deviceID);
+    int schemaRegionId = (int) (id >>> 32);
+    int autoIncrementID = (int) id;
+    if (schemaRegionId == -1) {
+      return deviceIdOfNonExistentDevice;
+    }
+    List<IDeviceID> deviceIDs = deviceIDsMap.get(schemaRegionId);
+    synchronized (deviceIDs) {
+      return (StandAloneAutoIncDeviceID) deviceIDs.get(autoIncrementID);
+    }
+  }
+
+  /**
+   * get device id from a device path
+   *
+   * @param devicePath device path, like: "root.sg.x.d1"
+   * @return a standAloneAutoIncDeviceID instance
+   */
+  private static StandAloneAutoIncDeviceID fromDevicePath(String devicePath) {
+    try {
+      // use idTable to determine whether the device has been created
+      IDTable idTable = IDTableManager.getInstance().getIDTable(new PartialPath(devicePath));
+      StandAloneAutoIncDeviceID deviceID = new StandAloneAutoIncDeviceID(devicePath);
+      if (idTable.getDeviceEntry(deviceID) != null) {
+        deviceID = (StandAloneAutoIncDeviceID) idTable.getDeviceEntry(deviceID).getDeviceID();
+        return deviceID;
+      } else {
+        return deviceIdOfNonExistentDevice;
+      }
+    } catch (IllegalPathException e) {
+      logger.info(e.getMessage());
+      return deviceIdOfNonExistentDevice;
+    }
+  }
+
+  /**
+   * get device id from a device path, if the device represented by the path does not exist, a
+   * StandAloneAutoIncDeviceID instance is generated for the path
+   *
+   * @param devicePath device path, like: "root.sg.x.d1"
+   * @return a standAloneAutoIncDeviceID instance
+   */
+  private static StandAloneAutoIncDeviceID buildDeviceID(String devicePath) {
+    try {
+      PartialPath path = new PartialPath(devicePath);
+      // use idTable to determine whether the device has been created
+      IDTable idTable = IDTableManager.getInstance().getIDTable(path);
+      StandAloneAutoIncDeviceID deviceID = new StandAloneAutoIncDeviceID(devicePath);
+      // this device is added for the first time
+      if (idTable.getDeviceEntry(deviceID) == null) {
+        SchemaRegionId schemaRegionId = configManager.getBelongedSchemaRegionId(path);
+        deviceID.schemaRegionId = schemaRegionId.getId();
```

Comment:
When there's need to create the deviceId, we should already know which schemaRegion it belongs rather than do computation again.